IQSS/dataverse
# Auxiliary File API Enhancements

**What this PR does / why we need it**:
This PR adds a few enhancements to the Aux file API:
- Auxiliary files can now also be associated with non-tabular files
- Improved error reporting
- The API will block attempts to create a duplicate auxiliary file
- Delete and list-by-origin calls have been added

**Which issue(s) this PR closes**:

Closes #8235

**Special notes for your reviewer**: 
- w.r.t. stopping duplicates - this is an existing bug - you can add two aux files with the same format params which then causes downloads to fail (as there isn't a single response to the query to find the file.)
- w.r.t. list-by-origin - this was all that QDR needed and I thought perhaps that requiring the origin to find files might be useful, but one could redesign to just list all aux files and require the caller to filter the results and/or to make the origin param optional (e.g. query param)
- w.r.t. non-tabular files: QDR is using aux files with document file types (Word, PDF) and I'm not sure there's a good reason to limit aux files to certain mime types. Hopefully not controversial to just remove the restriction
- w.r.t. error reporting - again hoping there's no controversy - just trying to report the various ways the add call can fail more clearly.

**Suggestions on how to test this**:
As usual, just following the doc examples should be good. Specific things to try would be
- try to add an aux file twice (same formatTag/formatVersion - doesn't have to be same exact file) and verify that the second call fails
- add an aux to a non-tabular file
- try delete
- try to list files from the origin you use in uploads

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**: api only

**Is there a release notes update needed for this change?**: draft provided - this is a tweak to an experimental API so probably nothing that needs to be prominent

**Additional documentation**:
